### ISSUE TITLE
conversor.js fails with scope -1

### ISSUE THREAD

**Issue by gregid** [ThiagoCComelli/JS2SHACL-JSON-Schema-to-SHACL-conversor#2]
Hi, nice project!

I've tested it on few schemas and I found it is failing on some with `RangeError: invalid array length` at https://github.com/ThiagoCComelli/JS2SHACL-JSON-Schema-to-SHACL-conversor/blob/master/conversor.js#L963

At this point it is trying to create an array with negative value for `scope = -1`.
Here is an example json schema that fails:
https://vega.github.io/schema/vega-lite/v5.json

**Comment by gregid** [comment]
Here is a minimal example of the fail (on boolean array):
```json
{
  "$ref": "#/definitions/TopLevelSpec",
  "$schema": "http://json-schema.org/draft-07/schema#",
  "definitions": {
    "DomainUnionWith": {
      "additionalProperties": false,
      "properties": {
        "unionWith": {
          "anyOf": [
            {
              "items": {
                "type": "number"
              },
              "type": "array"
            },
            {
              "items": {
                "type": "string"
              },
              "type": "array"
            },
            {
              "items": {
                "type": "boolean"
              },
              "type": "array"
            }
          ],
          "description": "Customized domain values to be union with the field's values or explicitly defined domain. Should be an array of valid scale domain values."
        }
      },
      "required": [
        "unionWith"
      ],
      "type": "object"
    }
  }
}
```

**Comment by ThiagoCComelli** [comment]
@gregid Thank you for your report.
This error can occur when it has an element (allOf, anyOf or oneOf).
But it was fixed and is working now, the shape may in some cases be badly Indentation, however it will be valid for validators/readers.

**Comment by gregid** [comment]
@ThiagoCComelli thank you for looking into this issue. Could you try running the following full schema:

https://vega.github.io/schema/vega-lite/v5.json

It fails on https://github.com/ThiagoCComelli/JS2SHACL-JSON-Schema-to-SHACL-conversor/blob/master/conversor.js#L556

**Comment by gregid** [comment]
The minimal example for this new error:
```json
{
  "$ref": "#/definitions/TopLevelSpec",
  "$schema": "http://json-schema.org/draft-07/schema#",
  "definitions": {
    "FieldOneOfPredicate": {
      "additionalProperties": false,
      "properties": {
        "field": {
          "$ref": "#/definitions/FieldName",
          "description": "Field to be tested."
        },
        "oneOf": {
          "anyOf": [
            {
              "items": {
                "type": "string"
              },
              "type": "array"
            },
            {
              "items": {
                "type": "number"
              },
              "type": "array"
            },
            {
              "items": {
                "type": "boolean"
              },
              "type": "array"
            },
            {
              "items": {
                "$ref": "#/definitions/DateTime"
              },
              "type": "array"
            }
          ],
          "description": "A set of values that the `field`'s value should be a member of, for a data item included in the filtered data."
        }
      },
      "required": [
        "field",
        "oneOf"
      ],
      "type": "object"
    }
  }
}
```
It looks like the issue is caused by:
```json
        "oneOf": {
          "anyOf": [
            {...
```
as the code expects oneOf to be an array but it receives an object instead. Looking at the [specification](https://json-schema.org/draft/2020-12/json-schema-core.html#rfc.section.10.2.1.3) it states:
```
10.2.1.3. oneOf
This keyword's value MUST be a non-empty array. Each item of the array MUST be a valid JSON Schema.

An instance validates successfully against this keyword if it validates successfully against exactly one schema defined by this keyword's value.
```
So it looks like the conversor code is correct and vega-lite-v5.json shema is incorrect in this regard.

**Comment by gregid** [comment]
By changing oneOf to be an array the conversion to SHACL was successful.

**Comment by gregid** [comment]
An update from vega:
https://github.com/vega/vega-lite/issues/7349

Apparently it should be treated as a property, therefore an object should be accepted.

**Comment by gregid** [comment]
Let me rephrase my last post.
At the moment properties with the same name as a json schema constraint (e.g. `oneOf`, `anyOf`, `allOf`, etc.) are being treated as constraints by conversor.

**Comment by ThiagoCComelli** [comment]
This is an interesting case that you found. When I developed this tool, I had not thought about this possibility of having properties with the same name as some JSON "protected" word (e.g. `anyOf`, `oneOf`, `allOf`).

I will probably think about the next few days to implement this possibility. Anyway, it seems counter intuitive to use JSON "protected" words to declare property names.
